### PR TITLE
ci: Nightly release builds

### PR DESCRIPTION
**Pull request by Inphi**
Nightly builds added to ci. Images are built for the `nightly` tag.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -317,3 +365,93 @@ workflows:
...
+          docker_file: ops/docker/js-builder/Dockerfile
+          docker_tags: ethereumoptimism/js-builder:latest
+          docker_context: .
+      - docker-publish:
+          name: proxyd-release
+          docker_file: go/proxyd/Dockerfile
+          docker_tags: ethereumoptimism/proxyd:latest
+          docker_context: .
+      - docker-publish:
+          name: lsgeth-exporter-release
```
Comment (optimisticben) [inline .circleci/config.yml:410]: Spelling? `l2geth-exporter-release`

**Review by optimisticben — APPROVED** [review]
Just a spelling issue, otherwise lgtm

**Comment by tynes** [comment]
Do we want to use the `latest` tag for nightly builds? Previously `latest` always meant the latest tagged release. What do you think about making the tag be a datetime + git short hash?

**Comment by tynes** [comment]
> > Do we want to use the `latest` tag for nightly builds? Previously `latest` always meant the latest tagged release. What do you think about making the tag be a datetime + git short hash?
> 
> Good point. In that case, I'd rather use a `nightly` tag. A datetime + git hash isn't easy to discover by, for example, the nightly test cluster.

Ah good point! nightly tag is a good idea

**Comment by tynes** [comment]
This looks good to me

**Review by tynes — APPROVED** [review]

**Pull request merged**